### ISSUE TITLE
fire edit/delete actions on custom column

### ISSUE THREAD

**Issue by steve-frank** [akveo/ng2-smart-table#559]
I have added a column with custom renderer like this, 
settings = {
        actions: false,
        columns: {
 operation:{
              title:"operation",
              type: 'custom',
              renderComponent: CustomActionRenderComponent,
          }
}

and now it looks like this : 

![image](https://user-images.githubusercontent.com/31277095/31214732-84446c54-a961-11e7-9ba2-686831b78f54.png)

I want to fire the default edit or delete action when user clicks on link in dropdown. 
Basically I want to call the same function which are invoked when default action column is enabled and links in that column are clicked.

**Comment by steve-frank** [comment]
trying to hack this more...my renderComponent class is CustomActionRenderComponent.
In this class I tried to import TbodyEditDeleteComponent, which renders the default the action column.

now my class looks like this : 

```
import { Component, Input, OnInit } from '@angular/core';

import { ViewCell } from 'ng2-smart-table';
import { TbodyEditDeleteComponent } from 'ng2-smart-table/components/tbody/cells/edit-delete.component';


@Component({
  template: `
   
        <li  ngbDropdown class="nav-item dropdown pointer" style="list-style:none" routerLinkActive="active" [routerLinkActiveOptions]="{exact: true}">
                <a class="nav-link dropdown-toggle" ngbDropdownToggle href="javascript:void(0);" >
                    <span>
                        <i class="fa fa-th-list" aria-hidden="true"></i>
                        <span>
                            Action
                        </span>
                        <b class="caret"></b>
                    </span>
                </a>

                <ul class="dropdown-menu" ngbDropdownMenu>
<div>
                    <li>
                        <a class="dropdown-item" (click)="onEdit($event)">
                            <i class="fa fa-fw fa-edit" aria-hidden="true"></i>
                            <span >Edit</span>
                        </a>
                    </li>
                    <li>
                         <a class="dropdown-item" >
                            <i class="fa fa-fw fa-trash" aria-hidden="true"></i>
                            <span>Delete</span>
                        </a>
                    </li>
</div>
                </ul>
        </li>
                    
 
  `,
})


export class CustomActionRenderComponent extends TbodyEditDeleteComponent  implements ViewCell, OnInit {

  renderValue: string;

  @Input() value: string | number;
  @Input() rowData: any;

  ngOnInit() {
    this.renderValue = this.value == null? "":this.value.toString();
  }

}
```

and as expected it complaints on  Cannot read property 'getSetting' of undefined. which is line no. 40 (  if (this.grid.getSetting('mode') === 'external') {) in edit-delete.component.ts in the component code. 

will exporting these members from this class help? and also may be adding to index.ts so that we dont need to import with explicit path? 

pls help..this will be a common use case.. where User will want custom UI to action column. or may be provide a one more config in setting which takes in the "template"  string as input.

**Comment by filipe-nunes** [comment]
+1
